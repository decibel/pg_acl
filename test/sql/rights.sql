\set ECHO none

\i test/pgxntool/setup.sql

-- TODO: remove
SET search_path=public,tap;

-- NOTE: the compat.sql test verifies that we've got the full set of rights

-- Sanity-check with grant version
SELECT is(
  length(_aclitems_all_rights())
  , length(_aclitems_all_rights_no_grant()) * 2
  , '_aclitems_all_rights is 2x longer than without.'
);
SELECT is(
  replace(_aclitems_all_rights(), '*', '')
  , _aclitems_all_rights_no_grant())
  , '_aclitems_all_rights with *s removed matches without grants'
);


-- Function output
SELECT is(
  rights_to_enum_no_grant(_aclitems_all_rights_no_grant())
  , _all_acl_right_no_grant()
  , 'rights_to_enum_no_grant() output correct'
);
SELECT is(
  rights_to_enum(_aclitems_all_rights())
  , _all_acl_right()
  , 'rights_to_enum() output correct'
);


-- Function return types should be different
SELECT function_returns(
  'rights_to_enum_no_grant'::name
  , '{text}'::text[]
  , 'acl_right_no_grant'
);
SELECT function_returns(
  'rights_to_enum'::name
  , '{text}'::text[]
  , 'acl_right'
);

\echo TRANSACTION INTENTIONALLY LEFT OPEN!

-- vi: expandtab sw=2 ts=2
