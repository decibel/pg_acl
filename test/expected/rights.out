\set ECHO none
1..11
ok 1 - _aclitems_all_rights_w_grant is 2x longer than without.
ok 2 - _aclitems_all_rights_w_grant with *s removed matches without grants
ok 3 - Verify all values are in _all__acl_right()
ok 4 - Verify proper ordering of rights without grant
ok 5 - Verify proper ordering of rights with grant
ok 6 - rights_to_enum_no_grant() output correct
ok 7 - rights_to_enum() output correct
ok 8 - Function rights_to_enum_no_grant(text) should return acl_right_no_grant[]
ok 9 - Function rights_to_enum_no_grant(text) should return acl_right_no_grant[]
ok 10 - Cast (acl_right AS acl_right_no_grant) should exist
ok 11 - Cast (acl_right_no_grant AS acl_right) should exist
# TRANSACTION INTENTIONALLY LEFT OPEN!
